src/scripts/contentCapture/pdfScreenshotHelper.ts

```diff
@@ -29,42 +30,40 @@ export class PdfScreenshotHelper {
 		return new Promise<PdfScreenshotResult>((resolve, reject) => {
 			let getBinaryEvent = new Log.Event.PromiseEvent(Log.Event.Label.GetBinaryRequest);
 
-			let request = new XMLHttpRequest();
-			request.open("GET", url, true);
-
-			request.responseType = "arraybuffer";
-
-			let errorCallback = (failureInfo: OneNoteApi.RequestError) => {
-				getBinaryEvent.setStatus(Log.Status.Failed);
-				failureInfo.response = failureInfo.responseHeaders = undefined;
-				getBinaryEvent.setFailureInfo(failureInfo);
-				Clipper.logger.logEvent(getBinaryEvent);
-				reject();
+			let errorCallback = (failureInfo: Promise<OneNoteApi.RequestError>) => {
+				failureInfo.then((error) => {
+					getBinaryEvent.setStatus(Log.Status.Failed);
+					error.response = error.responseHeaders = undefined;
+					getBinaryEvent.setFailureInfo(error);
+					Clipper.logger.logEvent(getBinaryEvent);
+					reject();
+				})
 			};
 
-			request.onload = () => {
-				if (request.status === 200 && request.response) {
-					let arrayBuffer = request.response;
 
+			fetch(url)
+			.then(response => {
+				if (!response.ok) {
+					errorCallback(ErrorUtils.createRequestErrorObject(response, OneNoteApi.RequestErrorType.UNEXPECTED_RESPONSE_STATUS));
```

Comment:
`createRequestErrorObject` now returns a Promise. Should we add a `then` block here?

Reply:
My rationale behind this is that we pass the Promise object directly into the errorCallback function and have a then block inside it to resolve the promise. Let me know if you see a problem with this approach.

Reply:
Got it, thanks!